fields/types/sortableRelateds/SortableRelateds.js

```diff
@@ -0,0 +1,298 @@
+    if (cachedValues.length === postIds.length) {
+      this.setState({ loading: false, value: this.props.many ? cachedValues : cachedValues[0] });
+      return;
+    }
+    this.setState({ loading: true, value: null });
+    async.map(postIds, (postId, done) => {
+      xhr({
+        url: Keystone.adminPath + '/api/' + this.props.refList.path + '/' + postId,
+        responseType: 'json',
+      }, (err, resp, data) => {
+        if (err || !data) return done(err);
+        this.cacheItem(data);
+        done(err, data);
+      });
+    }, (err, expanded) => {
+      if (!this.isMounted()) return;
+      this.setState({ loading: false, value: this.props.many ? expanded : expanded[0] });
+    });
+  },
+
+  loadOptions(postIds) {
+    const filters = this.buildFilters();
+    xhr({
+      url: Keystone.adminPath + '/api/' + this.props.refList.path + '?basic&search=' + '' + '&' + filters,
+      responseType: 'json',
+    }, (err, resp, data) => {
+      if (err) {
+        console.error('Error loading items:', err);
+        return;
+      }
+      data.results.forEach(post => {
+        if (post) {
+          this._options.push({ label: post.slug, value: post.id });
+          this.cacheItem(post);
+        }
+      });
+      postIds = Array.isArray(postIds) ? postIds : postIds.split(',');
+      this.setState({ options: this._options.filter(postOption => postOption && !postIds.includes(postOption.value)) });
+    });
+  },
+
+  updatePickUpStatus() {
+    const { value } = this.state;
+    if (!Array.isArray(value)) {
+      return;
+    }
+
+    let numPickedUp = 0;
+    value.forEach(post => {
+      if (post && post.isPickedUpToRemove) {
+        numPickedUp++;
+      }
+    });
+
+    let pickUpStatus = PickUp.INDETERMINATE;
+    if (numPickedUp === 0) {
+      pickUpStatus = PickUp.NONE;
+    } else if (numPickedUp === value.length) {
+      pickUpStatus = PickUp.ALL;
+    }
+
+    this.setState({ pickUpStatus: pickUpStatus });
+  },
+
+  onPickUpSingle(postId) {
+    const { value } = this.state;
+    const post = Array.isArray(value) ? value.find(post => post && post.id === postId) : undefined;
+    if (post) {
+      post.isPickedUpToRemove = !post.isPickedUpToRemove ? true : false;
+    }
+    this.setState({ value }, this.updatePickUpStatus);
+  },
+
+  onPickUpAll() {
+    const { value, pickUpStatus } = this.state;
+    if (!Array.isArray(value)) {
+      return;
+    }
+    const invertedStatus = pickUpStatus === PickUp.NONE ? PickUp.ALL : PickUp.NONE;
+    value.forEach(post => { post.isPickedUpToRemove = invertedStatus === PickUp.ALL; });
+    this.setState({
+      value: value,
+      pickUpStatus: invertedStatus
+    });
+  },
+
+  onPickedUpRemove() {
+    const { value, pickUpStatus } = this.state;
+    if (!Array.isArray(value) || pickUpStatus === PickUp.NONE) {
+      return;
+    }
+
+    const pickedUp = value.filter(post => post && post.isPickedUpToRemove);
```

Comment:
How about using [lodash.partition](https://lodash.com/docs/4.17.15#partition) to achieve the same goal without filter the array twice if we already have lodash in this repo ?

Maybe it could be like:

`const [ pickedUp, remained ] = _.partition(value, post => post && post.isPickedUpToRemove)`

Reply:
nice suggestion! updated.